Python/Product/Common/Infrastructure/ProcessOutput.cs

```diff
@@ -330,8 +326,37 @@ select connection.LocalEndPoint.Port
                 string.Join("|", env.Select(kv => kv.Key + "=" + Convert.ToBase64String(utf8.GetBytes(kv.Value)))) :
                 "";
 
-            listener.Start();
-            listener.AcceptTcpClientAsync().ContinueWith(t => {
+            TcpListener listener = null;
+            Task<TcpClient> clientTask = null;
+
+            for (int retries = 10; retries >= 0; --retries) {
+                int port = GetFreePort();
+                listener = new TcpListener(IPAddress.Loopback, port);
+                psi.Arguments = port.ToString();
+                try {
+                    listener.Start();
+                } catch (SocketException) {
+                    if (retries == 0) {
+                        throw;
+                    }
+                    continue;
+                }
+                try {
+                    clientTask = listener.AcceptTcpClientAsync();
```

Comment:
I believe it could occur on the initial call (which is synchronous and returns a completed task object if it's ready) or after awaiting - because this is a wrapper around an old-style async function (Begin/EndAccept) it looks like the exception may fall out.

But your point is well taken - we should be protecting the `t.Result` below too.